samples/subsys/zbus/benchmark/src/benchmark.c

```diff
@@ -197,18 +197,21 @@ static void producer_thread(void)
 
 		zbus_chan_notify(&bm_channel, K_MSEC(200));
 	}
-	uint32_t duration = (k_uptime_get_32() - start);
+	uint32_t end = sys_clock_cycle_get_32();
+	uint32_t duration = k_cyc_to_ns_near32(end) - k_cyc_to_ns_near32(start);
 
 	if (duration == 0) {
 		LOG_ERR("Something wrong. Duration is zero!\n");
+		LOG_ERR("start = %" PRIu32 "\n", start);
+		LOG_ERR("end = %" PRIu32 "\n", end);
 		k_oops();
 	}
-	uint64_t i = (BYTES_TO_BE_SENT * 1000) / duration;
-	uint64_t f = ((BYTES_TO_BE_SENT * 100000) / duration) % 100;
+	uint64_t i = (BYTES_TO_BE_SENT * NSEC_PER_SEC / MB(1)) / duration;
+	uint64_t f = (BYTES_TO_BE_SENT * NSEC_PER_SEC * 100 / MB(1) / duration) % 100;
```

Comment:
I would add parenthesis around `BYTES_TO_BE_SENT * NSEC_PER_SEC` and `BYTES_TO_BE_SENT * NSEC_PER_SEC * 100` just to make explicit the precedence I desire.

Reply:
Done!